osbs/core.py

```diff
@@ -601,14 +602,31 @@ def delete_resource_quota(self, name):
         return response
 
     def watch_resource(self, resource_type, resource_name=None, **request_args):
+        def log_and_sleep():
+            logger.debug("connection closed, reconnecting in %ds", WATCH_RETRY_SECS)
+            time.sleep(WATCH_RETRY_SECS)
+
         path = "watch/namespaces/%s/%s/" % (self.namespace, resource_type)
         if resource_name is not None:
             path += "%s/" % resource_name
         url = self._build_url(path, _prepend_namespace=False, **request_args)
 
+        bad_responses = 0
         for retry in range(WATCH_RETRY):
-            response = self._get(url, stream=True, headers={'Connection': 'close'})
-            check_response(response)
+            try:
+                response = self._get(url, stream=True, headers={'Connection': 'close'})
+                check_response(response)
+            # we're already retrying, so there's no need to panic just because of a bad response
+            except OsbsResponseException as exc:
+                bad_responses += 1
+                if bad_responses > MAX_BAD_RESPONSES:
```

Comment:
I really like https://pypi.org/project/retry/ which does exactly the same.

Unfortunately it is not packaged in Fedora/Centos. But I really like the way with decorators.

I see there is `retry_on_exception` in `osbs-client/osbs/utils.py`. Could be used that decorator here?

Reply:
retry_on_exception will retry the entire function.  That's overkill.  I just want to catch the exception and continue.

Reply:
ok